packages/plugins/src/foundry/index.ts

```diff
@@ -369,20 +430,138 @@ const command = args[0]
       const { parsedConfig, minimalParsedConfig, configArtifacts } =
         await readUnvalidatedParsedConfig(configPath, cre, getConfigArtifacts)
 
-      // TODO: We should write parsedConfig and configArtifacts to a 'chugsplash-cache' file or
-      // something. We'll need them in the next ffi call, `postParsingValidation`
+      const configCache = {
+        parsedConfig,
+        configArtifacts,
+      }
+
+      if (!fs.existsSync('./cache')) {
+        fs.mkdirSync('./cache')
+      }
+      fs.writeFileSync(
+        './cache/chugsplash-config-cache.json',
+        JSON.stringify(configCache, null, 2),
+        'utf-8'
+      )
+
+      const ChugSplashFoundryABI =
+        // eslint-disable-next-line @typescript-eslint/no-var-requires
+        require('../../out/artifacts/ChugSplash.sol/ChugSplash.json').abi
+      const minimalParsedConfigType = ChugSplashFoundryABI.find(
+        (fragment) => fragment.name === 'ffiGetMinimalParsedConfig'
+      ).outputs[0]
+
+      const encodedMinimalParsedConfig = ethers.utils.defaultAbiCoder.encode(
+        [minimalParsedConfigType],
+        [minimalParsedConfig]
+      )
+      process.stdout.write(encodedMinimalParsedConfig)
+      break
+    }
+    case 'postParsingValidation': {
+      const encodedConfigCache = args[1]
+      const configCache = decodeCachedConfig(encodedConfigCache)
+
+      const {
+        parsedConfig,
+        configArtifacts,
+      }: {
+        parsedConfig: ParsedChugSplashConfig
+        configArtifacts: ConfigArtifacts
+      } = JSON.parse(
+        (
+          await fs.readFileSync('./cache/chugsplash-config-cache.json')
+        ).toString()
+      )
+
+      const { canonicalConfigFolder } = await getPaths()
+      const cre = await createChugSplashRuntime(
+        '',
+        false,
+        true,
+        canonicalConfigFolder,
+        undefined,
+        false,
+        process.stdout
+      )
+
+      await postParsingValidation(
+        parsedConfig,
+        configArtifacts,
+        cre,
+        configCache,
+        true
+      )
+      break
+    }
+    case 'getCurrentChugSplashManagerVersion': {
+      const artifactStructABI =
+        'tuple(uint256 major, uint256 minor, uint256 patch)'
+      const encodedVersion = ethers.utils.AbiCoder.prototype.encode(
```

Comment:
nit: I think the syntax that Ethers encourages is `ethers.utils.defaultAbiCoder.encode`